airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -0,0 +1,486 @@
+                if not cursor_date or updated > cursor_date:
+                    send_cnt += 1
+                    yield record
+            if not send_cnt:
+                self._finished = True
+        yield from []
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+
+        max_updated_at = self.datetime2str(self._max_cursor_date) if self._max_cursor_date else ""
+        return {self.cursor_field: max(max_updated_at, (current_stream_state or {}).get(self.cursor_field, ""))}
+
+    @property
+    def is_finished(self):
+        return self._finished
+
+    @abstractmethod
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        """can be different for each case"""
+
+
+class IncrementalBasicUnsortedPageStream(IncrementalBasicUnsortedStream, ABC):
+    """basic stream for loading without sorting but with pagination
+    This logic can be used for a small data size when this data is loaded fast
+    """
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        next_page = self._parse_next_page_number(response)
+        if not next_page:
+            self._finished = True
+            return None
+        return next_page
+
+    def request_params(
+        self, stream_state: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state, next_page_token)
+        params["page"] = next_page_token or 1
+        return params
+
+
+class FullRefreshBasicStream(IncrementalBasicUnsortedPageStream, ABC):
+    """"Basic stream for endpoints where there are not any created_at or updated_at fields"""
+
+    state_checkpoint_interval = None
+    cursor_field = []
+
+
+class IncrementalBasicSortedCursorStream(IncrementalBasicUnsortedStream, ABC):
+    """basic stream for loading sorting data with cursor based pagination"""
+
+    def request_params(
+        self, stream_state: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state, next_page_token)
+        params.update({"sort_by": self.cursor_field, "sort_order": "desc", "limit": self.state_checkpoint_interval})
+
+        if next_page_token:
+            params["cursor"] = next_page_token
+        return params
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        if self.is_finished:
+            return None
+        return response.json().get("before_cursor")
+
+
+class IncrementalBasicSortedPageStream(IncrementalBasicUnsortedPageStream, ABC):
+    """basic stream for loading sorting data with normal pagination"""
+
+    def request_params(
+        self, stream_state: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs
```

Comment:
using named args in a list of kwargs usually means we want to do something with those args. however since we're not using `stream_state` and `next_page_token` args we can use `**kwargs` as the full signature and params for everything, no need to pass the params in one by one.

Reply:
I see. Done